app/src/main/java/com/arctouch/codechallenge/model/repository/retrofit/TmdbRepository.kt

```diff
@@ -0,0 +1,22 @@
+package com.arctouch.codechallenge.model.repository.retrofit
+
+import com.arctouch.codechallenge.BuildConfig
+import com.arctouch.codechallenge.model.data.GenreResponse
+import com.arctouch.codechallenge.model.data.Movie
+import com.arctouch.codechallenge.model.data.UpcomingMoviesResponse
+import retrofit2.Response
+
+class TmdbRepository(private val api: TmdbApi) {
+
+    suspend fun getMoviesGenre(page: Int): Response<GenreResponse> {
```

Comment:
I thought of two possible solutions for this: The first would be to make a call on the splash screen and save the list of genres in Room and fetch it according to the loading of the remote call list and the second would be to make a series of parallel  calls to load this when making the network call for the first time, in my opinion the first one would be more correct since we would not hold the user too long on the home screen.

Reply:
While there are many ways of overcoming this, your second idea is usually preferred over splash loading. You could have a cached repository that would either make the api request and save it to the cache or just return the cache if that's already present. More on that here: https://developer.android.com/jetpack/docs/guide#cache-data
Specific cases need to be handled separately (i. e. when you need sensitive information that might be outdated in the cache), but that's the general idea.